StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/OpenCloseSpacingCodeFixProvider.cs

```diff
@@ -92,7 +92,63 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                     triviaList = prevToken.TrailingTrivia.AddRange(token.LeadingTrivia);
 
                     replaceMap[prevToken] = prevToken.WithTrailingTrivia();
-                    replaceMap[token] = token.WithLeadingTrivia(triviaList.WithoutTrailingWhitespace());
+                    if (triviaList.All(i => i.IsKind(SyntaxKind.WhitespaceTrivia) || i.IsKind(SyntaxKind.EndOfLineTrivia)))
+                    {
+                        replaceMap[token] = token.WithLeadingTrivia();
+                    }
+                    else if (token.IsFirstInLine() && token.IsLastInLine())
+                    {
+                        /* This block covers the case where `token` is the only non-trivia token on its line. However,
+                         * the line may still contain non-whitespace trivia which we want the removal process to
+                         * preserve. This code fix only removes the whitespace surrounding `token` if it is the only
+                         * non-whitespace token on the line.
+                         */
+                        int lastNewLineLeading = token.LeadingTrivia.LastIndexOf(SyntaxKind.EndOfLineTrivia);
+                        int firstNewLineFollowing = token.TrailingTrivia.IndexOf(SyntaxKind.EndOfLineTrivia);
+                        bool onlyWhitespace = true;
+                        for (int i = lastNewLineLeading + 1; i < token.LeadingTrivia.Count; i++)
+                        {
+                            onlyWhitespace &= token.LeadingTrivia[i].IsKind(SyntaxKind.WhitespaceTrivia);
+                        }
+
+                        firstNewLineFollowing = firstNewLineFollowing == -1 ? token.TrailingTrivia.Count : firstNewLineFollowing;
+                        for (int i = 0; i < firstNewLineFollowing; i++)
+                        {
+                            onlyWhitespace &= token.TrailingTrivia[i].IsKind(SyntaxKind.WhitespaceTrivia);
+                        }
+
+                        if (onlyWhitespace)
+                        {
+                            // Move the token, and remove the other tokens from its line. Keep all other surrounding
+                            // trivia. Keep the last newline that precedes token, but not the first that follows it.
+                            SyntaxTriviaList trailingTrivia = prevToken.TrailingTrivia;
+                            if (lastNewLineLeading >= 0)
+                            {
+                                trailingTrivia.AddRange(token.LeadingTrivia.Take(lastNewLineLeading + 1));
+                            }
+                            else
+                            {
+                                trailingTrivia.AddRange(token.LeadingTrivia);
+                            }
+
+                            // firstNewLineFollowing was adjusted above to account for the missing case.
+                            trailingTrivia.AddRange(token.TrailingTrivia.Take(firstNewLineFollowing));
```

Comment:
I added new tests. If either of the fix two cases is not corrected, at least one of the new tests will fail. I haven't found a way to test the last case, but I fixed it.